agent/src/main/java/dev/aikido/agent/wrappers/spring/SpringWebfluxWrapper.java

```diff
@@ -0,0 +1,119 @@
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static net.bytebuddy.implementation.bytecode.assign.Assigner.Typing.DYNAMIC;
+import static net.bytebuddy.matcher.ElementMatchers.*;
+
+/**
+ * Wraps handle() function on HttpWebHandlerAdapter for Spring Webflux.
+ * Creates context object, writes a response (e.g. ip blocking), and reports status code.
+ * [github link](https://github.com/spring-projects/spring-framework/blob/7405e2069098400a01ee1e84ce72c45c6498b28d/spring-web/src/main/java/org/springframework/web/server/adapter/HttpWebHandlerAdapter.java#L269)
+ */
+public class SpringWebfluxWrapper implements Wrapper {
+    public static final Logger logger = LogManager.getLogger(SpringWebfluxWrapper.class);
+
+    @Override
+    public String getName() {
+        return SpringWebfluxAdvice.class.getName();
+    }
+
+    @Override
+    public ElementMatcher<? super MethodDescription> getMatcher() {
+
+        return isDeclaredBy(getTypeMatcher()).and(named("handle"))
+                .and(takesArgument(0, nameContains("ServerHttpRequest")))
+                .and(takesArgument(1, nameContains("ServerHttpResponse")));
+    }
+
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return nameContainsIgnoreCase("org.springframework.web.server.adapter.HttpWebHandlerAdapter");
+    }
+
+    public record SkipOnWrapper(Mono<Void> mono) {
+    }
+
+    public static class SpringWebfluxAdvice {
+        @Advice.OnMethodEnter(skipOn = SkipOnWrapper.class, suppress = Throwable.class)
+        public static Object onEnter(
+                @Advice.Origin Executable method,
+                @Advice.Argument(value = 0, typing = DYNAMIC, optional = true) ServerHttpRequest req,
+                @Advice.Argument(value = 1, typing = DYNAMIC, optional = true) ServerHttpResponse res
+        ) {
+            if (req == null) {
+                return null;
+            }
+            // Extract headers & query parameters :
+            Set<Map.Entry<String, List<String>>> headerEntries = req.getHeaders().entrySet();
+            Map<String, List<String>> query = req.getQueryParams();
+
+            // Extract cookies :
+            HashMap<String, List<String>> cookieMap = new HashMap<>();
+            for (Map.Entry<String, List<HttpCookie>> entry : req.getCookies().entrySet()) {
+                List<String> values = entry.getValue().stream().map(HttpCookie::getValue).collect(Collectors.toList());
+                cookieMap.put(entry.getKey(), values);
+            }
+
+            // Create context object :
+            ContextObject context = new SpringWebfluxContextObject(
+                    req.getMethod().toString(), req.getURI().toString(),
+                    Objects.requireNonNull(req.getRemoteAddress()),
+                    cookieMap, query, req.getHeaders().toSingleValueMap()
+            );
+
+            // If a response is present, write the response :
```

Comment:
why would a response already be ready on method enter? let's comment

Reply:
Updating comment...